http-common/src/request.rs

```diff
@@ -16,6 +16,41 @@ where
     request_with_headers(client, method, uri, None, body).await
 }
 
+pub async fn request_with_retry<TRequest, TResponse>(
+    client: &hyper::Client<super::Connector, hyper::Body>,
+    method: http::Method,
+    uri: &str,
+    body: Option<&TRequest>,
+    max_retries: u32,
+) -> std::io::Result<TResponse>
+where
+    TRequest: serde::Serialize,
+    TResponse: serde::de::DeserializeOwned,
+{
+    let mut retry_num = 0;
+    loop {
+        match request(client, method.clone(), uri, body).await {
+            Ok(response) => return Ok(response),
+            Err(err) => {
```

Comment:
Retry should only happen for throttling due to connection closed.

Reply:
is there a reason to not just always retry?

Reply:
Key does not exist already and needs to be created first - error.

Cert does not exist already and needs to be created first - error.

We had discussed already that we only want to retry the requests to host services in case of throttling. The only cases of retrying for all errors is with network services, ie Hub and DPS.

Reply:
ok, then in that case should I still call it max_retries and request_with_retry? I'm having trouble thinking of a good name for just retrying on throttling

Reply:
Names are fine. Just add a doc comment.

(Ideally Hub and DPS clients would also have used `http_common` instead of `hyper::Client::request` directly. Then `request_*_with_retry` could check what kind of `Connector` it is and behave accordingly. `Connector::Unix` => retry only on connection-closed, `Connector::Tcp` => retry on all errors.)